components/interface/TreeWidget.js

```diff
@@ -383,47 +382,97 @@ export default class TreeWidget extends React.Component {
+      }
+    }
+    return min;
+  }
+
   componentWillMount () {
     if (window.templateID !== undefined) {
       this.initTree(window.templateID);
+      document.addEventListener('mousedown', this.monitorMouseClick, false);
     }
   }
 
+  componentWillReceiveProps () {
+    document.removeEventListener('mousedown', this.monitorMouseClick, false);
+  }
+
   nodeClick (event, rowInfo) {
     console.log("clicked on the tree node");
     console.log(rowInfo);
     this.selectNode(rowInfo.node);
   }
 
+  monitorMouseClick (e) {
+    if (this.colorPickerNode !== undefined && this.colorPickerNode.contains(e.target)) {
+      return;
+    } else {
+      this.colorPickerNode = undefined;
+      this.setState({ displayColorPicker: false });
+    }
+  }
+
   getButtons (rowInfo) {
     var buttons = [];
     if (rowInfo.node.title !== "No data available.") {
-      buttons.push(<Tooltip 
-        title={rowInfo.node.description}
-        style={{ fontSize: "14px" }}>
-        <i className="fa fa-info-circle"
-          aria-hidden="true"
-          onClick={ () => {
-            window.addVfbId(rowInfo.node.instanceId);
-          }}></i></Tooltip>);
+      buttons.push(<i className="fa fa-tint"
+        aria-hidden="true"
+        onClick={ () => {
+          this.setState({ displayColorPicker: true });
+        }}>
+        { (this.state.displayColorPicker 
+          && this.state.nodeSelected.subtitle === rowInfo.node.subtitle
+          && this.colorPickerNode === undefined) 
+          ? <CompactColor 
+            style={{ zIndex: 10 }}/>
+          : null}
+      </i>);
     }
     return buttons;
   }
 
   getNodes (rowInfo) {
     if (rowInfo.node.title !== "No data available.") {
-      var title = <div className={rowInfo.node.isSelected ? "nodeSelected" : "nodeUnselected"}>
-        {rowInfo.node.title}
-      </div>;
+      var title = <MuiThemeProvider theme={this.theme}>
+        <Tooltip
+          title={<div id="dario">
+            <div> {rowInfo.node.instanceId} </div>
+            <div> {rowInfo.node.description} </div>
+            <div>
+              <img id="dario"
+                src={"https://VirtualFlyBrain.org/reports/" + rowInfo.node.instanceId + "/thumbnailT.png"} />
+            </div></div>}>
+          <div 
+            className={rowInfo.node.subtitle === this.state.nodeSelected.subtitle
+              ? "nodeFound nodeSelected" 
+              : "nodeSelected"}
+            onClick={ () => {
+              this.colorPickerNode = undefined;
+              this.props.selectionHandler(rowInfo.node.subtitle);
+              this.setState({ displayColorPicker: false }); 
+            }}>
+            {rowInfo.node.title}
+          </div>
+        </Tooltip>
+      </MuiThemeProvider>;
     }
     return title;
   }
 
   searchDone (matches) {
     if (matches.length > 0) {
-      matches.map(item => {
-        item.isSelected = true;
-      })
+      console.log('Found children in the tree');
```

Comment:
@ddelpiano what's up here? remove the commented code and the console log? Seems like a leftover